app/models/landing_page/block/statistics.rb

```diff
@@ -30,15 +32,21 @@ def rows
       rows
     end
 
+    def attachment
+      @attachment ||= landing_page.attachments.find { |att| att.filename == data["csv_file"] }
+    end
+
   private
 
     def csv_rows
-      rows = CSV.read(csv_file_path, headers: true).map(&:to_h)
-      rows.each(&:deep_symbolize_keys!)
-    end
-
-    def csv_file_path
-      @csv_file_path ||= Rails.root.join("#{STATISTICS_DATA_PATH}/#{data['csv_file']}")
+      rows = if attachment
+               CSV.new(URI.parse(attachment.url).open, headers: true).map(&:to_h)
+             else
+               # SCAFFOLDING
+               csv_file_path = Rails.root.join("#{STATISTICS_DATA_PATH}/#{data['csv_file']}")
+               CSV.read(csv_file_path, headers: true).map(&:to_h)
+             end
+      @csv_rows = rows.each(&:deep_symbolize_keys!)
```

Comment:
Minor, but I don't think you need to assign a `@csv_rows` instance variable here.

Reply:
I've adjusted this code slightly to memoize the calls to load/fetch the csv.